src/utils/SiriusAdapter.cpp

```diff
@@ -0,0 +1,309 @@
+
+    // needed for counting
+    int number = getIntOption_("number");
+    number = number + 1; //needed to write the correct number of compounds
+
+    // Parameter for Sirius3
+    QString executable = getStringOption_("executable").toQString();
+    QString profile = getStringOption_("profile").toQString();
+    QString elements = getStringOption_("elements").toQString();
+    QString database = getStringOption_("database").toQString();
+    QString isotope = getStringOption_("isotope").toQString();
+    QString noise = QString::number(getIntOption_("noise"));
+    QString ppm_max = QString::number(getIntOption_("ppm_max"));
+    QString candidates = QString::number(getIntOption_("candidates"));
+
+    bool auto_charge = getFlag_("auto_charge");
+    bool no_recalibration = getFlag_("no_recalibration");
+    bool fingerid = getFlag_("fingerid");
+    bool iontree = getFlag_("iontree");
+
+    //-------------------------------------------------------------
+    // Calculations
+    //-------------------------------------------------------------
+
+    PeakMap spectra;
+    MzMLFile f;
+    f.setLogType(log_type_);
+    f.load(in, spectra);
+    std::vector<String> subdirs;
+
+    //Write msfile
+    String ms_file = SiriusMSFile::store(spectra);
+    String tmp_dir = QDir::toNativeSeparators(String(File::getTempDirectory()).toQString()) + "/" + String(File::getUniqueName()).toQString();
+    String out_dir = tmp_dir + "/" + "sirius_out";
+
+    //Start Sirius
+    QStringList process_params; // the actual process
+    process_params << "-p" << profile
+                   << "-e" << elements
+                   << "-d" << database
+                   << "-s" << isotope
+                   << "--noise" << noise
+                   << "--candidates" << candidates
+                   << "--ppm-max" << ppm_max
+                   << "--quiet"
+                   << "--output" << out_dir.toQString(); //internal output folder for temporary
+
+    if (no_recalibration)
+    {
+      process_params << "--no-recalibration";
+    }
+    if (fingerid)
+    {
+      process_params << "--fingerid";
+    }
+    if (iontree)
+    {
+      process_params << "--iontree";
+    }
+    if (auto_charge)
+    {
+      process_params << "--auto-charge";
+    }
+
+    process_params << ms_file.toQString();
+
+    QProcess qp;
+    qp.start(executable, process_params); // does automatic escaping etc... start
+    bool success = qp.waitForFinished(-1); // wait till job is finished
+
+    if (success == false || qp.exitStatus() != 0 || qp.exitCode() != 0)
+    {
+      qp.close();
+      writeLog_( "Fatal error: Running SiriusAdapter returned an error code or could no compute the input" );
+    }
+
+    //-------------------------------------------------------------
+    // writing output
+    //-------------------------------------------------------------
+
+    //Extract path to subfolders (sirius internal folder structure)
```

Comment:
I think this should stay in the SiriusAdapter, since it is adapted to the temporary data structure of Sirius (tmp_folder and msFile). Internally the the functions of the file class are used. (removeDir (s.o - moved to file class) & remove)).